scripts/import-yoga-tests/src/main.rs

```diff
@@ -0,0 +1,78 @@
+use std::collections::HashSet;
+use std::fs;
+use std::path::{Path, PathBuf};
+
+struct YogaFixture {
+    #[allow(dead_code)]
+    file_name: PathBuf,
+    name: String,
+    content: String,
+}
+
+fn main() {
+    // Get Taffy fixtures dir
+    let script_root_dir = PathBuf::from(std::env::var("CARGO_MANIFEST_DIR").unwrap());
+    let taffy_repo_root = script_root_dir.parent().and_then(Path::parent).unwrap();
+    let taffy_fixtures_dir = taffy_repo_root.join("test_fixtures");
+
+    // Get Taffy fixture names
+    let taffy_fixture_names = fs::read_dir(&taffy_fixtures_dir)
+        .unwrap()
+        .filter_map(|a| a.ok())
+        .map(|f| {
+            let file_name = f.file_name();
+            let file_name = if file_name.to_string_lossy().starts_with('x') {
+                file_name.to_string_lossy()[1..].replace(".html", "")
+            } else {
+                file_name.to_string_lossy().replace(".html", "")
+            };
+            file_name
+        })
+        .collect::<HashSet<String>>();
+
+    // Get Taffy fixture template
+    let fixture_template_path = script_root_dir.join("FIXTURE_TEMPLATE.html");
+    let fixture_template = fs::read_to_string(fixture_template_path).unwrap();
+
+    // Get Yoga fixtures dir
+    let yoga_fixtures_dir =
+        PathBuf::from(std::env::var("YOGA_FIXTURE_DIR").expect("YOGA_FIXTURE_DIR env var must be set"));
+    let fixtures_file_list = fs::read_dir(yoga_fixtures_dir).unwrap();
+
+    let yoga_fixtures: Vec<YogaFixture> = fixtures_file_list
+        .filter_map(|a| a.ok())
+        .filter(|f| f.path().is_file() && f.path().extension().map(|p| p == "html").unwrap_or(false))
+        .flat_map(|f| {
+            let file_name: PathBuf = f.file_name().into();
+
+            let file_content = fs::read_to_string(f.path()).unwrap();
+
+            file_content
+                .split("\n\n")
+                .map(|snippet| {
+                    let snippet = snippet.trim();
+                    let first_quote_idx = snippet.chars().position(|c| c == '"').unwrap();
+                    let name: String = snippet.chars().skip(first_quote_idx + 1).take_while(|c| *c != '"').collect();
+                    let name_replace_snippet = snippet.replace(&name, "test-root");
+
+                    let taffy_name = {
+                        let name = name.replace("row_gap", "gap_row_gap");
+                        name.replace("column_gap", "gap_column_gap")
+                    };
+                    YogaFixture { file_name: file_name.clone(), name: taffy_name, content: name_replace_snippet }
+                })
+                .collect::<Vec<_>>()
+        })
+        .collect();
+
+    for fixture in yoga_fixtures {
+        if !taffy_fixture_names.contains(&fixture.name) {
```

Comment:
What happens if the fixtures are updated but not renamed? Won't they get out of sync with this strategy?

Reply:
> Won't they get out of sync with this strategy?

Yes. They likely already are. This is at least a quick way to get us tests we were entirely missing though. I guess we'll need to actually parse the HTML/CSS to sync updates.

Reply:
Makes sense. What's the cost to rebuilding them completely every time?

Reply:
> What's the cost to rebuilding them completely every time?

As in treating Yoga's version as the source of truth and overwriting any changes we might have made? There's no technical blocker on that. I think the main problem with that would be that it would make it hard for us to maintain any changes that we actually need. As an example: I removed whitespace from a couple of tests when I added support for sizing text nodes. It would be nice if there was a single source of truth though, and I suppose that could be managed by just being selective with when we sync and which changes we commit when we sync.

Reply:
Sounds reasonable: I'm fine with whatever you pick here.